### PR TITLE
Compat AWSS3 0.7 and 0.8

### PR DESCRIPTION
**Pull request by BSnelling**
Adding compat for AWSS3 0.7.  This required a change in one function for compatibility.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ Memento = "f28f55f0-a522-5efc-85c2-fe41dfb9b2d9"
 
 [compat]
 AWSCore = "0.5.5, 0.6"
-AWSS3 = "0.6"
+AWSS3 = "0.6, 0.7"
```
[the remark this answers is not in the document]
Comment (rofinn) [reply Project.toml:15]: Also, using 0.8.3 will likely fix the CI issue. It just hasn't been backported yet.

[the remark this answers is not in the document]
Comment (rofinn) [reply Project.toml:15]:
```suggestion
AWSS3 = "0.6, 0.7, 0.8"
```

**Comment by rofinn** [comment]
Backport for `AWSS3@0.7.7` finished, should be good to merge after the changes mentioned above.

File: src/handler.jl [hunk 2]
```diff
@@ -33,7 +33,7 @@ function path(handler::Handler{P}, name::String; tags...) where P
     parts = split(name, '.')
     parts[end] = string(parts[end], ".jlso")
 
-    return join(handler.path, prefix..., parts...)
+    return joinpath(handler.path, prefix..., parts...)
```
[the remark this answers is not in the document]
Comment (BSnelling) [reply src/handler.jl]: Makes sense, I've added to `src/Checkpoints.jl` and restored `join` here

**Comment by BSnelling** [comment]
Thank you for the review and the backporting.  This will now close #5 and #7

**Pull request merged**